docs/asciidoc/advanced-contextPropagation.adoc

```diff
@@ -32,25 +48,80 @@ TL.set("HELLO");
+      .map(v -> "delayed ctx[" + TLKEY + "]=" + ctx.getOrDefault(TLKEY, "not found") + ", TL=" + TL.get()))
+.contextCapture() // can be skipped in automatic mode when a blocking operator follows
+.block(); // returns "delayed ctx[TLKEY]=HELLO, TL=null" in default mode
+          // returns "delayed ctx[TLKEY]=HELLO, TL=HELLO" in automatic mode
+----
+====
+
+NOTE: In the **automatic** mode, blocking operators, such as `Flux#blockFirst()`,
+`Flux#blockLast()`, `Flux#toIterable()`, `Mono#block()`, `Mono#blockOptional()`, and
+relevant overloads, all perform `contextCapture()` transparently, so in most cases it is
+not necessary to add it.
+
+=== `contextWrite` Operator
+
+When the `ThreadLocal` values are not and do not need to be present at the time of
+subscription, we can immediately store the value in the `Context`:
+
+====
+[source,java]
+----
+//assuming TL is known to Context-Propagation as key TLKEY.
+static final ThreadLocal<String> TL = new ThreadLocal<>();
+
+//in the main thread, TL is not set
+
+Mono.deferContextual(ctx ->
+  Mono.delay(Duration.ofSeconds(1))
+      //we're now in another thread, TL is not explicitly set
+      .map(v -> "delayed ctx[" + TLKEY + "]=" + ctx.getOrDefault(TLKEY, "not found") + ", TL=" + TL.get()))
+.contextWrite(ctx -> ctx.put(TLKEY, "HELLO"))
+.block(); // returns "delayed ctx[TLKEY]=HELLO, TL=null" in default mode
+          // returns "delayed ctx[TLKEY]=HELLO, TL=HELLO" in automatic mode
 ----
 ====
 
-== Operators that transparently restore a snapshot: `handle` and `tap`
-Both `Flux` and `Mono` variants of `handle` and `tap` will have their behavior slightly modified
-if the Context-Propagation library is available at runtime.
+== Accessing `ThreadLocal` state corresponding to `ContextView`
+
+As can be seen in the above examples, in the default mode, the `ThreadLocal` value is
+`null` inside the lambda provided to the `map` operator. This shows a major difference
+between the **default** (limited) mode, and the **automatic** mode.
+
+=== Automatic mode
+
+In the **automatic** mode all operators restore `ThreadLocal` state across `Thread`
+boundaries. In contrast, in the **default** mode only selected operators do so.
 
-Namely, if their downstream `ContextView` is not empty they will assume a context capture has occurred
-(either manually or via the `contextCapture()` operator) and will attempt to restore `ThreadLocal`s from
-that snapshot transparently.
+It is not an easy task to achieve, as the Reactive Streams specification makes reactive
+chains `Thread`-agnostic. However, we did our best to control sources of `Thread` switches
+and perform snapshot restoration based on the Reactor `Context`, which is treated as the
+source of truth for `ThreadLocal` state. In case you are interested, we described how
+we got here in a three-article series on
+https://spring.io/blog/2023/03/28/context-propagation-with-project-reactor-1-the-basics[the spring.io blog].
```

Comment:
there was not any blog links before although there were dedicated blogs on those topics, thus maybe it is better to follow the same style as before and give brief explanation while a reader can find other sources on their own